### PR TITLE
Fix return value from deserialization

### PR DESCRIPTION
**Pull request by lordjim666**
Added different methods for converting object and array.

### CODE REVIEW

File: src/DataConverters/DefaultMysqlDataConverter.php [hunk 1]
```diff
@@ -287,19 +289,37 @@ public static function toArray($value) {
...
     }
 
     /**
-     * @param mixed $value
-     * @return mixed
+     * @param string $value
+     * @return object|null
+     */
+    public static function deserializeObject($value) {
+        if (!\is_string($value)) throw new \obo\Exceptions\Exception("");
```
Comment (lordjim666) [inline src/DataConverters/DefaultMysqlDataConverter.php]: Im not sure why we wanted that exception... and when

File: src/DataConverters/DefaultMysqlDataConverter.php [hunk 2]
```diff
@@ -287,19 +289,37 @@ public static function toArray($value) {
...
+    }
+
+    /**
+     * @param string $value
+     * @return array|mixed
      */
-    public static function deserialize($value) {
-        return \unserialize($value);
+    public static function deserializeArray($value) {
+        if (!\is_string($value)) throw new \obo\Exceptions\Exception("");
```
Comment (lordjim666) [inline src/DataConverters/DefaultMysqlDataConverter.php]: same here...

File: src/DataConverters/DefaultMysqlDataConverter.php (hunk 1 above)
Comment (hAdamm) [reply src/DataConverters/DefaultMysqlDataConverter.php]: because the notice can be turned off on the server

Comment (hAdamm) [reply src/DataConverters/DefaultMysqlDataConverter.php]: implementation should also take into account the "nullable" attribute of property

Comment (lordjim666) [reply src/DataConverters/DefaultMysqlDataConverter.php]: makes no sense imho

**Pull request merged**